source/extensions/transport_sockets/alts/tsi_socket.cc

```diff
@@ -106,14 +106,20 @@ Network::PostIoAction TsiSocket::doHandshakeNextDone(NextResultPtr&& next_result
     }
     if (handshake_validator_) {
       std::string err;
-      const bool peer_validated = handshake_validator_(peer, err);
+      TsiInfo tsi_info;
+      const bool peer_validated = handshake_validator_(peer, tsi_info, err);
       if (peer_validated) {
         ENVOY_CONN_LOG(debug, "TSI: Handshake validation succeeded.", callbacks_->connection());
       } else {
         ENVOY_CONN_LOG(debug, "TSI: Handshake validation failed: {}", callbacks_->connection(),
                        err);
         return Network::PostIoAction::Close;
       }
+      ProtobufWkt::Struct dynamic_metadata;
+      ProtobufWkt::Value val;
+      val.set_string_value(tsi_info.name_);
+      dynamic_metadata.mutable_fields()->insert({std::string("peer_name"), val});
+      callbacks_->connection().streamInfo().setDynamicMetadata("envoy.alts.tsi", dynamic_metadata);
```

Comment:
document this dynamic metadata in ALTS docs?

Reply:
alts don't have its own rst file, only a proto. Am I supposed to document it its proto file?

Reply:
Yes, I think adding a line in a proto file should be fine.